Index: resources/schema/request_log.schema.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- resources/schema/request_log.schema.json	(revision e93f68fc0b6f2144b831defa24d09897e3cf1cd2)
+++ resources/schema/request_log.schema.json	(revision )
@@ -7,9 +7,9 @@
 	},
 	"mappings": {
 		"request_log": {
-			"_source": {
-				"compress": true
-			},
+//			"_source": {
+//				"compress": true
+//			},
 			"dynamic_templates": [
 				{
 					"default_strings": {
@@ -36,10 +36,10 @@
 					"type": "string"
 				},
 				"query": {
-					"type": "object",
-					"enabled": false,
-					"index": false,
-					"store": true
+					"type": "string",
+//					"enabled": false,
+					"index": "no"
+//					"store": true
 				},
 				"data": {
 					"index": "not_analyzed",
@@ -47,10 +47,10 @@
 
 				},
 				"error": {
-					"type": "object",
-					"enabled": false,
-					"index": false,
-					"store": true
+					"type": "string",
+//					"enabled": false,
+					"index": "no"
+//					"store": true
 
 				},
 				"from": {
Index: resources/config/elasticsearch.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- resources/config/elasticsearch.yml	(revision e93f68fc0b6f2144b831defa24d09897e3cf1cd2)
+++ resources/config/elasticsearch.yml	(revision )
@@ -25,7 +25,7 @@
 
 index.number_of_shards: 1
 index.number_of_replicas: 1
-index.cache.field.type: soft
+# index.cache.field.type: soft
 index.translog.interval: 60s
 index.translog.flush_threshold_size: 1gb
 
Index: active_data/actions/save_query.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- active_data/actions/save_query.py	(revision e93f68fc0b6f2144b831defa24d09897e3cf1cd2)
+++ active_data/actions/save_query.py	(revision )
@@ -171,9 +171,9 @@
     "settings": {
         "index.number_of_shards": 3,
         "index.number_of_replicas": 2,
-        "index.store.throttle.type": "merge",
-        "index.cache.filter.expire": "1m",
-        "index.cache.field.type": "soft",
+        "index.store.throttle.type": "merge"
+        # "index.cache.filter.expire": "1m"
+        # "index.cache.field.type": "soft",
     },
     "mappings": {
         "_default_": {
@@ -184,7 +184,7 @@
                         "match_mapping_type": "string",
                         "mapping": {
                             "type": "string",
-                            "index": "not_analyzed"
+                            "index": "no"
                         }
                     }
                 }
@@ -198,24 +198,24 @@
             "properties": {
                 "create_time": {
                     "type": "double",
-                    "index": "not_analyzed",
-                    "store": True
+                    "index": "not_analyzed"
+                    # "store": True
                 },
                 "last_used": {
                     "type": "double",
-                    "index": "not_analyzed",
-                    "store": True
+                    "index": "not_analyzed"
+                    # "store": True
                 },
                 "hash": {
                     "type": "string",
-                    "index": "not_analyzed",
-                    "store": True
+                    "index": "not_analyzed"
+                    # "store": True
                 },
                 "query": {
-                    "type": "object",
-                    "enabled": False,
-                    "index": False,
-                    "store": True
+                    "type": "string",
+                    # "enabled": False,
+                    "index": "no"
+                    # "store": True
                 }
             }
         }
